src/transforms/reduce/merge_strategy.rs

```diff
@@ -132,6 +164,108 @@ impl ReduceValueMerger for ArrayMerger {
 
 //------------------------------------------------------------------------------
 
+#[derive(Debug, Clone)]
+struct LongestArrayMerger {
+    v: Vec<Value>,
+}
+
+impl LongestArrayMerger {
+    fn new(v: Vec<Value>) -> Self {
+        Self { v }
+    }
+}
+
+impl ReduceValueMerger for LongestArrayMerger {
+    fn add(&mut self, v: Value) -> Result<(), String> {
+        if let Value::Array(a) = v {
+            if a.len() > self.v.len() {
+                self.v = a;
+            }
+        }
+        Ok(())
+    }
+
+    fn insert_into(self: Box<Self>, k: String, v: &mut LogEvent) -> Result<(), String> {
+        v.insert(k, Value::Array(self.v));
+        Ok(())
+    }
+}
+
+//------------------------------------------------------------------------------
+
+#[derive(Debug, Clone)]
+struct ShortestArrayMerger {
+    v: Vec<Value>,
+}
+
+impl ShortestArrayMerger {
+    fn new(v: Vec<Value>) -> Self {
+        Self { v }
+    }
+}
+
+impl ReduceValueMerger for ShortestArrayMerger {
+    fn add(&mut self, v: Value) -> Result<(), String> {
+        if let Value::Array(a) = v {
+            if a.len() < self.v.len() {
+                self.v = a;
+            }
+        }
+        Ok(())
+    }
+
+    fn insert_into(self: Box<Self>, k: String, v: &mut LogEvent) -> Result<(), String> {
+        v.insert(k, Value::Array(self.v));
+        Ok(())
+    }
+}
+
+//------------------------------------------------------------------------------
+#[derive(Debug, Clone)]
+struct UniqueMerger {
+    v: HashSet<Value>,
+}
+
+fn insert_value(h: &mut HashSet<Value>, v: Value) {
+    match v {
+        Value::Map(m) => {
+            for (_, v) in m {
+                h.insert(v);
+            }
+        }
+        Value::Array(vec) => {
+            for v in vec {
+                h.insert(v);
+            }
+        }
+        _ => {
+            h.insert(v);
+        }
+    }
+}
+
+impl UniqueMerger {
+    fn new(v: Value) -> Self {
+        let mut h = HashSet::default();
```

Comment:
Small note that you might want to consider parameterizing over a different hash function than default.

Reply:
Any suggestion on which hash function?

Reply:
IIRC we already have twox-hash in the project but the exact choice would need to be guided by a benchmark. I'd be satisfied with an issue to follow up on that and you're welcome to assign it to me.